Add troubleshooting guide for eventing (#8362)
kyma-project/kyma: docs/event-mesh/10-01-troubleshooting.md

---
title: Troubleshooting
---

## Events do not reach the Namespace with active Eventing

> The upgrade of the Event Sources Controller Manager triggers the new Pod to update the KService for the HTTPSource. As a result of changes introduced to the KService, the Knative Serving Controller (KSC) creates a new Revision. The KSC does not pick up this new Revision and marks the KService as `Not ready`, still looking for the old Revision.

Reviewer: The upgrade of the Event Sources Controller Manager triggers the update of the KService for the HTTPSource. As a result of changes introduced to the KService, the Knative Serving Controller (KSC) creates a new Revision. The KSC does not pick up this new Revision and marks the KService as `Not ready`, while looking for the old Revision.

To fix this issue, delete the existing KService. It will be recreated automatically, pointing to the right Revision.

Follow these steps:

1. Check the status of the HTTP Source:

    ```bash
    $ kubectl get httpsources.sources.kyma-project.io -n kyma-integration

>      NAME          READY   REASON

Reviewer: There is an alignment issue with `READY`. I think there is one space extra before `READY`

    application   False   ServiceNotReady
    ```

2. Check the status of the corresponding KService:

    ```bash
    $ kubectl get ksvc -n kyma-integration
    NAME          URL                                                     LATESTCREATED          LATESTREADY          READY   REASON
    application   http://application.kyma-integration.svc.cluster.local   application-g4qd8      application-c2zlz    False   RevisionMissing
    ```

> 3. If the PrinterColumn `READY` displays the value `False`, and `REASON` shows `RevisionMissing`, delete the existing KService:

Reviewer: (nit): and the `REASON`

Author: not necessarily :) if we had the `REASON` PrinterColumn, then `the`would be necessary. I would go without or maybe I will rephrase it a bit.

>     ```bas

Reviewer: `bash`

    $ kubectl delete ksvc -n kyma-integration application
    ```

> 4. The KService will recreate automatically, pointing to the right revision. Check the status of the KService and the HTTP Source:

Reviewer: The KService will be recreated automatically

And there is no space in `HTTPSource`

---
title: Troubleshooting
---

## Events do not reach the Namespace

> 

Reviewer 2: or "No events in the Namespace" but I don't have any brilliant alternative :(

> The upgrade of the Event Sources Controller Manager triggers the new Pod to update the KService for the HTTPSource custom resource. As a result of changes introduced to the KService, the Knative Serving Controller (KSC) creates a new Revision. The KSC does not pick up this new Revision and, while looking for the old Revision, marks the KService as `Not ready`. This brakes the event flow and prevents events from reaching the Namespace.

Reviewer 2: ```suggestion
The upgrade of the Event Sources Controller Manager triggers the new Pod to update the KService connected with the HTTPSource custom resource (CR). As a result of changes introduced in the KService, the Knative Serving Controller (KSC) creates a new Revision. The KSC does not pick up this new Revision and marks the KService as `Not ready`, looking for the original Revision. This breaks the event flow and prevents events from reaching the Namespace.
```

Reviewer 2: If you still prefer `brakes` that's ok.

> To fix this issue, delete the existing KService. It will be recreated automatically, pointing to the right Revision.

Reviewer 2: ```suggestion
To fix this issue, delete the existing KService. It will be recreated automatically, pointing to the correct Revision.
```

Follow these steps:

1. Check the status of the HTTPSource CR:

    ```bash

>     $ kubectl get httpsources.sources.kyma-project.io -n kyma-integration

Reviewer 2: We rather don't use $ in command samples.

    
    NAME          READY   REASON
    application   False   ServiceNotReady
    ```

2. Check the status of the corresponding KService:

    ```bash
    $ kubectl get ksvc -n kyma-integration
    NAME          URL                                                     LATESTCREATED          LATESTREADY          READY   REASON
    application   http://application.kyma-integration.svc.cluster.local   application-g4qd8      application-c2zlz    False   RevisionMissing
    ```

3. If the values for the `READY` and `REASON` PrinterColumns are `False` and `RevisionMissing` respectively, delete the existing KService:

    ```bash
    $ kubectl delete ksvc -n kyma-integration application
    ```

> 4. The KService will be recreated automatically, pointing to the right revision. Check the status of the KService and the HTTPSource CR:

Reviewer 2: ```suggestion
4. The KService will be recreated automatically, pointing to the correct revision. Check the statuses of the KService and the HTTPSource CR:
```

>     ```bash

Reviewer 2: Could we separate it into 2 blocks?
